pkg/frontend/querymiddleware/limits.go

```diff
@@ -99,7 +105,8 @@ func (l limitsMiddleware) Do(ctx context.Context, r Request) (Response, error) {
 				"msg", "skipping the execution of the query because its time range is before the 'max query lookback' setting",
 				"reqStart", util.FormatTimeMillis(r.GetStart()),
 				"redEnd", util.FormatTimeMillis(r.GetEnd()),
-				"maxQueryLookback", maxQueryLookback)
+				"maxQueryLookback", maxQueryLookback,
+				"minBlockRetentionPeriod", blockRetentionPeriod)
 
 			return newEmptyPrometheusResponse(), nil
 		}
```

Comment:
Below there's a log:

> "the start time of the query has been manipulated because of the 'max query lookback' setting"

Few comments:
1. Update the message to mention the retention too
2. Add both `maxQueryLookback` and `minBlockRetentionPeriod` to the log